### ISSUE TITLE
blaspp and lapackpp are in our public API

### ISSUE THREAD

**Issue by teonnik** [eth-cscs/DLA-Future#232]
Parameters such as `blas::Op`, `blas::Uplo`, `blas::Side`, `lapack::Norm` are present in our API. That makes them a required dependency even if a different LAPACK/BLAS back end is used. If we want other back ends without requiring blaspp/lapackpp, we might want to provide our own API for the aforementioned parameters.

**Comment by rasolca** [comment]
You mean aliasing them to dlaf::Op, ..., to simplify the change of the backend?

**Comment by teonnik** [comment]
Yes, essentially defining our own equivalents such that we disentangle the back-ends from the public API.

**Comment by albestro** [comment]
In principles, I completely agree with this, I did your same thought when we started using them.

It is something we quickly talked about at the beginning of the project with @rasolca, and I remember that there was also discussion about using other wrappers (or writing new ones).
